src/dalle_mini/model/wandb_pretrained.py

```diff
@@ -0,0 +1,20 @@
+import os
+import wandb
+
+
+class PretrainedFromWandbMixin:
```

Comment:
I think you need to inherit from something HF related here

Reply:
Why? It's just a mixin that acts as a container for a function that would have to be replicated in the three classes otherwise. In the case of `DalleBartTokenizer`, the call to `super` [here](https://github.com/borisdayma/dalle-mini/pull/121/commits/7e48337ace14ef727e9ec502a82e15d0e5d26443#diff-b15e0ef27d313162d0611d09a85bb4671fe945e3c0cc4c01421e9e67c4ef0784R18) is resolved to `BartTokenizer.from_pretrained` per [this declaration](https://github.com/borisdayma/dalle-mini/pull/121/commits/7e48337ace14ef727e9ec502a82e15d0e5d26443#diff-3fbba2dcad5d89a1b2c8d2b64cf37165329282237e6cda57e46999c85ee26b85R10), whereas in the config it is resolved to `PretrainedConfig.from_pretrained` because of [this](https://github.com/borisdayma/dalle-mini/pull/121/commits/7e48337ace14ef727e9ec502a82e15d0e5d26443#diff-862ff738e4ff854ecec9c3e4fdfbad9c00de07ba75f213d74669fb98861aaae9R26), and so on. This implementation worked fine in my tests, in exactly the same way as when the function was copied to the three classes.

I'm still learning idiomatic Python for these tasks, but I've seen this technique being used frequently in the huggingface codebase, so I thought it was ok.

Is there a simpler solution that I'm missing? (other than copying the same function in the three implementations) Or a test case I didn't consider?

This could actually be used by anyone wanting to extend their code with wandb checkpoints, and not just in the context of our project.

Reply:
Oh that's great, I hadn't realized it was used in conjunction with another class!